### ISSUE TITLE
deployment of crowdsec-lapi fails when upgrading from 0.13.0 to 0.14.0

### ISSUE THREAD

**Issue by pfaelzerchen** [crowdsecurity/helm-charts#199]
I'm running on the helm chart version 0.13.0. When I try to deploy an upgrade to 0.14.0, the crowdsec-lapi pod isn't able to come up. The log file says:

```
no configuration for Central API (CAPI) (/etc/crowdsec/config.yaml)
```

I did no changes to my values.yaml so far:
```yaml
# -- for raw logs format: json or cri (docker|containerd)
container_runtime: containerd

# Here you can specify your own custom configuration to be loaded in crowdsec agent or lapi
# Each config needs to be a multi-line using '|' in YAML specs
# for the agent those configs will be loaded : parsers, scenarios, postoverflows, simulation.yaml
# for the lapi those configs will be loaded : profiles.yaml, notifications, console.yaml
config:
  # -- to better understand how to write a postoverflow, you can take a look at (https://docs.crowdsec.net/docs/next/whitelist/create/#whitelist-in-postoverflows)
  postoverflows:
    s01-whitelist:
      mydyndns-whitelist.yaml: |
        name: my/dyndnswhitelist ## Must be unqiue
        description: "Whitelist events from my own dynamic IP"
        whitelist:
          reason: "My dynamic IP"
          expression:
              - evt.Overflow.Alert.Source.IP in LookupHost("test.net")

tls:
  enabled: true
  bouncer:
    reflector:
      namespaces: ["traefik"]

# lapi will deploy pod with crowdsec lapi and dashboard as deployment
lapi:
  # -- environment variables from crowdsecurity/crowdsec docker image
  env:
    # by default disable the agent because it only needs the local API.
    # - name: DISABLE_ONLINE_API
    #   value: "true"
    - name: ENROLL_KEY
      valueFrom:
        secretKeyRef:
          name: crowdsec-credentials
          key: enroll-key
    - name: ENROLL_INSTANCE_NAME
      value: "Homecluster"
    - name: ENROLL_TAGS
      value: "k3s linux home"

  resources:
    limits:
      memory: 350Mi
    requests:
      cpu: 15m
      memory: 350Mi

  # -- Enable persistent volumes
  persistentVolume:
    # -- Persistent volume for data folder. Stores e.g. registered bouncer api keys
    data:
      enabled: true
      accessModes:
        - ReadWriteOnce
      storageClassName: ceph-block
      size: 5Gi
    # -- Persistent volume for config folder. Stores e.g. online api credentials
    config:
      enabled: true
      accessModes:
        - ReadWriteOnce
      storageClassName: ceph-block
      size: 100Mi

  # -- Enable service monitoring (exposes "metrics" port "6060" for Prometheus)
  metrics:
    enabled: true
    # -- Creates a ServiceMonitor so Prometheus will monitor this service
    # -- Prometheus needs to be configured to watch on all namespaces for ServiceMonitors
    # -- See the documentation: https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#prometheusioscrape
    # -- See also: https://github.com/prometheus-community/helm-charts/issues/106#issuecomment-700847774
    serviceMonitor:
      enabled: true
      additionalLabels: {}

# agent will deploy pod on every node as daemonSet to read wanted pods logs
agent:
  acquisition:
    # -- Specify each pod you want to process it logs (namespace, podName and program)
    # Traefik
    - namespace: "traefik"
      # -- to select pod logs to process
      podName: "traefik-*"
      # -- program name related to specific parser you will use (see https://hub.crowdsec.net/author/crowdsecurity/configurations/docker-logs)
      program: "traefik" #nginx
      # -- If set to true, will poll the files using os.Stat instead of using inotify
      poll_without_inotify: false
    # Authelia
    - namespace: "authelia"
      podName: "authelia-*"
      program: "authelia"

  resources:
    limits:
      memory: 350Mi
    requests:
      cpu: 15m
      memory: 350Mi

  # -- Enable persistent volumes
  persistentVolume:
    # -- Persistent volume for config folder. Stores local config (parsers, scenarios etc.)
    config:
      enabled: true
      accessModes:
        - ReadWriteMany
      storageClassName: ceph-filesystem
      existingClaim: ""
      size: 100Mi

  # -- environment variables from crowdsecurity/crowdsec docker image
  env:
    # by default we configure the docker-logs parser to be able to parse docker logs in k8s
    # by default we disable local API on the agent pod
    # - name: SCENARIOS
    #   value: "scenario/name otherScenario/name"
    # - name: PARSERS
    #   value: "parser/name otherParser/name"
    # - name: POSTOVERFLOWS
    #   value: "postoverflow/name otherPostoverflow/name"
    # - name: CONFIG_FILE
    #   value: "/etc/crowdsec/config.yaml"
    # - name: DSN
    #   value: "file:///var/log/toto.log"
    # - name: TYPE
    #   value: "Labels.type_for_time-machine_mode"
    # - name: TEST_MODE
    #   value: "false"
    # - name: TZ
    #   value: ""
    # - name: DISABLE_AGENT
    #   value: "false"
    # - name: DISABLE_ONLINE_API
    #   value: "false"
    # - name: LEVEL_TRACE
    #   value: "false"
    # - name: LEVEL_DEBUG
    #   value: "false"
    # - name: LEVEL_INFO
    #   value: "false"
    - name: PARSERS
      value: "crowdsecurity/cri-logs"
    - name: COLLECTIONS
      value: "crowdsecurity/traefik LePresidente/authelia"
    # When testing, allow bans on private networks
    # - name: DISABLE_PARSERS
    #   value: "crowdsecurity/whitelists"

  # -- Enable service monitoring (exposes "metrics" port "6060" for Prometheus)
  metrics:
    enabled: true
    # -- Creates a ServiceMonitor so Prometheus will monitor this service
    # -- Prometheus needs to be configured to watch on all namespaces for ServiceMonitors
    # -- See the documentation: https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#prometheusioscrape
    # -- See also: https://github.com/prometheus-community/helm-charts/issues/106#issuecomment-700847774
    serviceMonitor:
      enabled: true
      additionalLabels: {}
```

The agent pods come up without trouble.

Any idea, where I could look further into? Or is it a bug?

**Comment by blotus** [comment]
Hello,

Can you provide the entire log output from the LAPI pod ?

Can you also share the content of `/etc/crowdsec/config.yaml` from the volume mounted in the LAPI pod ?

We have changed the way we handle CAPI credentials to not require a persistent volume anymore to persists them, and we had no issue upgrading clusters during the testing phase, but it seems we have missed a corner case.

**Comment by srkoster** [comment]
I'm facing the same error.

LAPI pod log output
```
Populating configuration directory...
sending incremental file list
acquis.yaml
config.yaml
console.yaml
dev.yaml
local_api_credentials.yaml
online_api_credentials.yaml
simulation.yaml
user.yaml
acquis.d/
appsec-configs/
appsec-rules/
collections/
collections/linux.yaml -> /etc/crowdsec/hub/collections/crowdsecurity/linux.yaml
collections/sshd.yaml -> /etc/crowdsec/hub/collections/crowdsecurity/sshd.yaml
console/
console/context.yaml
contexts/
contexts/bf_base.yaml -> /etc/crowdsec/hub/contexts/crowdsecurity/bf_base.yaml
hub/
hub/.index.json
hub/collections/
hub/collections/crowdsecurity/
hub/collections/crowdsecurity/linux.yaml
hub/collections/crowdsecurity/sshd.yaml
hub/contexts/
hub/contexts/crowdsecurity/
hub/contexts/crowdsecurity/bf_base.yaml
hub/parsers/
hub/parsers/s00-raw/
hub/parsers/s00-raw/crowdsecurity/
hub/parsers/s00-raw/crowdsecurity/syslog-logs.yaml
hub/parsers/s01-parse/
hub/parsers/s01-parse/crowdsecurity/
hub/parsers/s01-parse/crowdsecurity/sshd-logs.yaml
hub/parsers/s02-enrich/
hub/parsers/s02-enrich/crowdsecurity/
hub/parsers/s02-enrich/crowdsecurity/dateparse-enrich.yaml
hub/parsers/s02-enrich/crowdsecurity/geoip-enrich.yaml
hub/parsers/s02-enrich/crowdsecurity/whitelists.yaml
hub/scenarios/
hub/scenarios/crowdsecurity/
hub/scenarios/crowdsecurity/ssh-bf.yaml
hub/scenarios/crowdsecurity/ssh-cve-2024-6387.yaml
hub/scenarios/crowdsecurity/ssh-slow-bf.yaml
notifications/
notifications/email.yaml
notifications/sentinel.yaml
notifications/slack.yaml
notifications/splunk.yaml
parsers/
parsers/s00-raw/
parsers/s00-raw/syslog-logs.yaml -> /etc/crowdsec/hub/parsers/s00-raw/crowdsecurity/syslog-logs.yaml
parsers/s01-parse/
parsers/s01-parse/sshd-logs.yaml -> /etc/crowdsec/hub/parsers/s01-parse/crowdsecurity/sshd-logs.yaml
parsers/s02-enrich/
parsers/s02-enrich/dateparse-enrich.yaml -> /etc/crowdsec/hub/parsers/s02-enrich/crowdsecurity/dateparse-enrich.yaml
parsers/s02-enrich/geoip-enrich.yaml -> /etc/crowdsec/hub/parsers/s02-enrich/crowdsecurity/geoip-enrich.yaml
parsers/s02-enrich/whitelists.yaml -> /etc/crowdsec/hub/parsers/s02-enrich/crowdsecurity/whitelists.yaml
patterns/
patterns/aws
patterns/bacula
patterns/bro
patterns/cowrie_honeypot
patterns/exim
patterns/firewalls
patterns/haproxy
patterns/java
patterns/junos
patterns/linux-syslog
patterns/mcollective
patterns/modsecurity
patterns/mongodb
patterns/mysql
patterns/nagios
patterns/nginx
patterns/paths
patterns/postgresql
patterns/rails
patterns/redis
patterns/ruby
patterns/smb
patterns/ssh
patterns/tcpdump
postoverflows/
scenarios/
scenarios/ssh-bf.yaml -> /etc/crowdsec/hub/scenarios/crowdsecurity/ssh-bf.yaml
scenarios/ssh-cve-2024-6387.yaml -> /etc/crowdsec/hub/scenarios/crowdsecurity/ssh-cve-2024-6387.yaml
scenarios/ssh-slow-bf.yaml -> /etc/crowdsec/hub/scenarios/crowdsecurity/ssh-slow-bf.yaml
sent 811,961 bytes  received 1,104 bytes  1,626,130.00 bytes/sec
total size is 810,010  speedup is 1.00
Error: no matches found
Generate local agent credentials
Machine 'crowdsec-lapi-76677cd6cc-8rm4q' successfully added to the local API.
API credentials written to '/etc/crowdsec/local_api_credentials.yaml'.
Check if lapi needs to register an additional agent
Machine 'zxqv2mgdxUWf8O5eVhIF0b8RXcgKBi5aCaF2CvQkcNr22UQe' successfully added to the local API.
API credentials written to '/dev/null'.
Agent registered to lapi
level=fatal msg="no configuration for Central API (CAPI) in '/etc/crowdsec/config.yaml'"
```

There's no `/etc/crowdsec/config.yaml` volume mounted in the LAPI pod as far as I can tell. Here's the pod YAML file

```
apiVersion: v1
kind: Pod
metadata:
  name: crowdsec-lapi-76677cd6cc-8rm4q
  namespace: crowdsec
spec:
  containers:
    - env:
        - name: LOCAL_API_URL
          value: http://localhost:8080
        - name: AGENT_USERNAME
          valueFrom:
            secretKeyRef:
              key: username
              name: agent-credentials
        - name: AGENT_PASSWORD
          valueFrom:
            secretKeyRef:
              key: password
              name: agent-credentials
        - name: DISABLE_AGENT
          value: 'true'
        - name: INSECURE_SKIP_VERIFY
          value: 'false'
        - name: CS_LAPI_SECRET
          valueFrom:
            secretKeyRef:
              key: csLapiSecret
              name: crowdsec-lapi-secrets
        - name: REGISTRATION_TOKEN
          valueFrom:
            secretKeyRef:
              key: registrationToken
              name: crowdsec-lapi-secrets
        - name: CUSTOM_HOSTNAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        - name: DISABLE_ONLINE_API
          value: 'false'
        - name: ENROLL_KEY
          valueFrom:
            secretKeyRef:
              key: crowdsec_enroll_key
              name: crowdsec-credentials
        - name: ENROLL_INSTANCE_NAME
          value: Talos_Cluster
        - name: ENROLL_TAGS
          value: homelab
        - name: DISABLE_AGENT
          value: 'true'
      image: crowdsecurity/crowdsec:v1.6.3
      imagePullPolicy: IfNotPresent
      livenessProbe:
        failureThreshold: 3
        httpGet:
          path: /health
          port: lapi
          scheme: HTTP
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
      name: crowdsec-lapi
      ports:
        - containerPort: 8080
          name: lapi
          protocol: TCP
        - containerPort: 6060
          name: metrics
          protocol: TCP
      readinessProbe:
        failureThreshold: 3
        httpGet:
          path: /health
          port: lapi
          scheme: HTTP
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
      resources:
        limits:
          cpu: 500m
          memory: 500Mi
        requests:
          cpu: 500m
          memory: 500Mi
      securityContext:
        allowPrivilegeEscalation: false
        privileged: false
      startupProbe:
        failureThreshold: 30
        httpGet:
          path: /health
          port: lapi
          scheme: HTTP
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      volumeMounts:
        - mountPath: /docker_start.sh
          name: crowdsec-docker-start-script
          subPath: docker_start.sh
        - mountPath: /staging/etc/crowdsec/online_api_credentials.yaml
          name: crowdsec-capi-credentials-volume
          subPath: online_api_credentials.yaml
        - mountPath: /etc/crowdsec/profiles.yaml
          name: crowdsec-profiles-volume
          subPath: profiles.yaml
        - mountPath: /etc/crowdsec/config.yaml.local
          name: crowdsec-config-local-volume
          subPath: config.yaml.local
        - mountPath: /etc/crowdsec/notifications/http.yaml
          name: crowdsec-notifications-http
          subPath: http.yaml
        - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
          name: kube-api-access-f5qjg
          readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: talos-worker-03
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
    - effect: NoExecute
      key: node.kubernetes.io/not-ready
      operator: Exists
      tolerationSeconds: 300
    - effect: NoExecute
      key: node.kubernetes.io/unreachable
      operator: Exists
      tolerationSeconds: 300
  volumes:
    - configMap:
        defaultMode: 420
        name: crowdsec-docker-start-script-configmap
      name: crowdsec-docker-start-script
    - name: crowdsec-capi-credentials-volume
      secret:
        defaultMode: 420
        secretName: crowdsec-capi-credentials
    - configMap:
        defaultMode: 420
        name: crowdsec-profiles
      name: crowdsec-profiles-volume
    - configMap:
        defaultMode: 420
        name: crowdsec-config-local
      name: crowdsec-config-local-volume
    - configMap:
        defaultMode: 420
        items:
          - key: http.yaml
            path: http.yaml
        name: crowdsec-notifications
      name: crowdsec-notifications-http
    - name: kube-api-access-f5qjg
      projected:
        defaultMode: 420
        sources:
          - serviceAccountToken:
              expirationSeconds: 3607
              path: token
          - configMap:
              items:
                - key: ca.crt
                  path: ca.crt
              name: kube-root-ca.crt
          - downwardAPI:
              items:
                - fieldRef:
                    apiVersion: v1
                    fieldPath: metadata.namespace
                  path: namespace
```

**Comment by blotus** [comment]
@srkoster can you try to set `lapi.storeCAPICredentialsInSecret` to `true` in your values ?
This should make crowdsec regenerate the credentials (if you had no persistent volume for the LAPI config before this release, crowdsec would always regenerate CAPI credentials on start, which is never what you want)

**Comment by srkoster** [comment]
@blotus I already have set lapi.storeCAPICredentialsInSecret to true. Also, I have disabled all lapi persistence, have disabled tls and am using postgresql as my database.

I'll post my values.yaml later today for reference.

**Comment by Darkfella91** [comment]
Hey, im having the same issue after updating the chart's version. I tried the proposed solution by disabling persistence and setting storeCAPICredentialsInSecret to true. Before that i also had problem with my custom profiles.yaml and notification's http.yaml config files. LAPI pod was complaining about not being able to copy them over cos they already exist, so i had to temporary comment them out in my helmrelease but then this issue came up. Here is my helmrelease file for reference. 

```apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: crowdsec
spec:
  interval: 30m
  chart:
    spec:
      chart: crowdsec
      version: 0.14.0
      sourceRef:
        kind: HelmRepository
        name: crowdsec
        namespace: flux-system
  maxHistory: 3
  install:
    remediation:
      retries: 3
  upgrade:
    cleanupOnFail: true
    remediation:
      strategy: rollback
      retries: 3

  values:
    config:

      #profiles.yaml: |
        #name: default_ip_remediation
          #debug: true
          #filters:
          #- Alert.Remediation == true && Alert.GetScope() == "Ip"
          #decisions:
          #- type: ban
            #duration: 4h
          #duration_expr: Sprintf('%dh', (GetDecisionsCount(Alert.GetValue()) + 1) * 4)
          #notifications:
            #- pushover
          #on_success: break
          #---
          #name: default_range_remediation
          #debug: true
          #filters:
          #- Alert.Remediation == true && Alert.GetScope() == "Range"
          #decisions:
          #- type: ban
            #duration: 4h
          #duration_expr: Sprintf('%dh', (GetDecisionsCount(Alert.GetValue()) + 1) * 4)
          #notifications:
            #- pushover
          #on_success: break

      #notifications:
        #http.yaml: |
          #type: http
          #name: pushover

          #log_level: info

          #group_wait: "30s"
          #group_threshold: 10
          #max_retry: 5
          #timeout: "10s"

          #-------------------------
          # plugin-specific options

          # The following template receives a list of models.Alert objects
          # The output goes in the http request body
          #format: |
            #{
            #"token": $$CROWDSEC_PUSHOVER_TOKEN,
            #"user": $$PUSHOVER_USER_KEY,
            #"message": "{{`{{range . -}}{{$alert := . -}}{{range .Decisions -}}<b>{{.Value}}</b> will get <b>{{.Type}}</b> for the next <b>{{.Duration}}</b> for triggering <b>{{.Scenario}}</b>.\r\n https://www.shodan.io/host/{{.Value}}{{end -}}{{end -}}`}}",
            #"html": "1",
            #"title": "Scenario triggered on IDS/IPS !"
            #}
          #url: https://api.pushover.net/1/messages.json
          #method: POST
          #headers:
            #Content-Type: "application/json"

      config.yaml.local: |
        db_config:
          type:     postgresql
          user:     $${DB_USER}
          password: $${DB_PASSWORD}
          db_name:  crowdsec
          host:     postgres17-rw.database.svc.cluster.local
          port:     5432
          sslmode:  require

    container_runtime: containerd

    agent:
      wait_for_lapi:
        image:
          tag: "1.37.0@sha256:768e5c6f5cb6db0794eec98dc7a967f40631746c32232b78a3105fb946f3ab83"

      acquisition:
        - namespace: network
          podName: nginx-external-controller-*
          program: nginx

        - namespace: media
          podName: jellyseerr-*
          program: jellyseerr

        - namespace: vaultwarden
          podName: vaultwarden-*
          program: VAULTWARDEN

        - namespace: idp
          podName: keycloak-0
          program: keycloak

      env:
        - name: LOCAL_API_URL
          value: "https://crowdsec-api.${PUBLIC_DOMAIN}"

        - name: PARSERS
          value: "crowdsecurity/cri-logs crowdsecurity/whitelists crowdsecurity/geoip-enrich"

        - name: COLLECTIONS
          value: "crowdsecurity/modsecurity inherent-io/keycloak LePresidente/jellyseerr crowdsecurity/nginx Dominic-Wagner/vaultwarden crowdsecurity/base-http-scenarios crowdsecurity/http-cve crowdsecurity/http-dos crowdsecurity/whitelist-good-actors"

      metrics:
        enabled: true
        serviceMonitor:
          enabled: true

    lapi:
      deployAnnotations:
        reloader.stakater.com/auto: "true"

      extraInitContainers:
        - name: initdb
          image: "ghcr.io/onedr0p/postgres-init:16.5@sha256:53fa938cfe6d2dae23b120145362341950f1154bf9d0e7f2ec4e471d61624f79"
          imagePullPolicy: IfNotPresent

          envFrom:
            - secretRef:
                name: &secret crowdsec-secret

      persistentVolume:
        data:
          enabled: false

        config:
          enabled: false

      storeCAPICredentialsInSecret: true

      ingress:
        enabled: true

        annotations:
          nginx.ingress.kubernetes.io/backend-protocol: "HTTP"
          external-dns.alpha.kubernetes.io/target: internal.${PUBLIC_DOMAIN}
          cert-manager.io/cluster-issuer: zerossl-prod

        ingressClassName: internal

        host: &host "crowdsec-api.${PUBLIC_DOMAIN}"

        tls:
          - hosts:
            - *host
            secretName: crowdsec-tls

      env:
        - name: CROWDSEC_PUSHOVER_TOKEN
          valueFrom:
            secretKeyRef:
              name: *secret
              key: CROWDSEC_PUSHOVER_TOKEN

        - name: PUSHOVER_USER_KEY
          valueFrom:
            secretKeyRef:
              name: *secret
              key: PUSHOVER_USER_KEY

        - name: DISABLE_ONLINE_API
          value: "false"

        - name: ENROLL_KEY
          valueFrom:
            secretKeyRef:
              name: *secret
              key: ENROLL_KEY

        - name: ENROLL_INSTANCE_NAME
          value: "cluster"

        - name: DB_PASSWORD
          valueFrom:
            secretKeyRef:
              name: *secret
              key: INIT_POSTGRES_PASS

        - name: DB_USER
          valueFrom:
            secretKeyRef:
              name: *secret
              key: INIT_POSTGRES_USER

        - name: BOUNCER_KEY_NGINX_EXTERNAL
          valueFrom:
            secretKeyRef:
              name: *secret
              key: BOUNCER_KEY_NGINX_EXTERNAL

      metrics:
        enabled: true
        serviceMonitor:
          enabled: true
```

**Comment by blotus** [comment]
Hello,

I've managed to reproduce the issue locally.

It's caused by the `ENROLL_KEY` env var in the LAPI pod configuration.

If you remove it, the pod should start fine. I'm starting to work on a fix.

**Comment by srkoster** [comment]
Good to hear that you've found the issue. I can confirm that my pods can start when I remove the `ENROLL_KEY` env var in the LAPI pod configuration. However, removing it removes the capability to enroll the engine into the online console. Waiting for your patch and hope that you'll find a solution for the enrollment.

**Comment by blotus** [comment]
Hello,

I've just released version 0.14.1 which should fix the issue.
Please let me know if you still encounter any errors with it.

**Comment by pfaelzerchen** [comment]
I can confirm that the fix worked. Thanks for your effort.
